core/authenticate/service.go

```diff
@@ -156,11 +166,43 @@ func (s Service) StartFlow(ctx context.Context, request RegistrationStartRequest
 		FinishURL: request.ReturnToURL,
 		CreatedAt: s.Now(),
 		ExpiresAt: s.Now().Add(defaultFlowExp),
+		Email:     request.Email,
 		Metadata: metadata.Metadata{
 			"callback_url": request.CallbackUrl,
 		},
 	}
 
+	if request.Method == PassKeyAuthMethod.String() {
+		newPassKeyUser := strategy.NewPassKeyUser(request.Email)
+		needRegistration := false
+		loggedInUser, err := s.userService.GetByID(ctx, request.Email)
+		if err != nil {
+			needRegistration = true
+		} else {
+			storedPasskey, passKeyExists := loggedInUser.Metadata["credentials"]
+			if !passKeyExists {
+				needRegistration = true
+			}
+			if _, ok := storedPasskey.(string); !ok {
+				needRegistration = true
+			}
+		}
+
+		if needRegistration {
+			response, err := s.startPassKeyRegisterMethod(ctx, newPassKeyUser, flow)
+			if err != nil && !errors.Is(err, ErrStrategyNotApplicable) {
```

Comment:
I don't see `ErrStrategyNotApplicable` being returned by `startPassKeyRegisterMethod`? Same with login, then why do we check it here?

Reply:
changed to just error